orchestra/communication/staffing.py

```diff
@@ -21,20 +22,22 @@ def handle_staffing_response(worker, staffing_request_id, is_available=False):
     response = StaffingResponse.objects.filter(request=staffing_request)
     if response.exists():
         response = response.first()
+        if not is_available and response.is_winner:
+            raise StaffingResponseException(
+                'Cannot reject after accepting the task')
+
         response.is_available = is_available
 
-        if not is_available and response.is_winner:
-            remove_worker_from_task(
-                worker.user.username,
-                staffing_request.task.id)
-            response.is_winner = False
     else:
         response = StaffingResponse.objects.create(
             request=staffing_request,
             is_available=is_available)
 
     if (is_available and
-            not StaffingResponse.objects.filter(is_winner=True).exists()):
+            not StaffingResponse.objects.filter(
+                    request__task=staffing_request.task,
+                    request__required_role=staffing_request.required_role,
```

Comment:
how does this work for multiple levels of review?  do we need a required_role_level as well?